vast_csi/plugins/block.py

```diff
@@ -479,6 +486,21 @@ def NodeStageVolume(
         device_path = device.DevicePath
         change_io_policy(device_name=device.Name, io_policy="round-robin")
 
+        host_encryption_present = any(key.startswith("host_encryption.") for key in volume_context)
+        if host_encryption_present:
+            host_encryption_secret = getattr(exit_stack, "host_encryption_secret", None)
+
+            if host_encryption_secret:
+                passphrase = host_encryption_secret
```

Comment:
I'm a bit confused about this if/else

passphrase =  host_encryption_secret so first "if host_encryption_secret" should do the trick. Why check again  "if not passphrase:" <- Seems like this condition never happen